### PR TITLE
Fixed Get Adjustment and related test

### LINKED ISSUE #149: Adjustments.Get returns uninitialized object
I am using version 1.3.1.
When I call this code (I don't see in you public Library that this code has changed):

```
    public class Adjustments
    {
        public static Adjustment Get(string uuid)
        {
            var adjustment = new Adjustment();
            Client.Instance.PerformRequest(Client.HttpRequestMethod.Get,
                "/adjustments/" + Uri.EscapeUriString(uuid),
                adjustment.WriteXml);
            return adjustment;
        }
    }

```

the returned object is not initialized. Also there is no check on the return code on the PerformRequest.  It seems this should use ReadXml instead?? and also check the return code like:

```
        public static Invoice Get(string invoiceNumberWithPrefix)
        {
            var invoice = new Invoice();
            
            var statusCode = Client.Instance.PerformRequest(Client.HttpRequestMethod.Get,
                Invoice.UrlPrefix + invoiceNumberWithPrefix,
                invoice.ReadXml);

            return statusCode == HttpStatusCode.NotFound ? null : invoice;
        }

```

### PR DESCRIPTION
**Pull request by MuffinMan10110**
fixes #149

### CODE REVIEW

**Comment by bhelx** [comment]
Thanks @MuffinMan10110 ! I've been out of town. Should get this merged today or tomorrow.

File: Library/Coupon.cs [hunk 1]
```diff
@@ -71,7 +71,7 @@ public enum CouponType
         public CouponDiscountType DiscountType { get; private set; }
         public CouponState State { get; private set; }
         public RedemptionResourceType RedemptionResource { get; set; }
-        public CouponType Type { get; set; } = CouponType.SingleCode;
```
Comment (bhelx) [inline Library/Coupon.cs]: @MuffinMan10110 what is the purpose of this change and what are it's implications?

Comment (MuffinMan10110) [inline Library/Coupon.cs]: @bhelx I didn't realize that is C# 6.0 syntax for setting the default value for a property. It didn't compile for me and I thought I had accidentally put it there so I removed it. My change should be reverted.

**Comment by MuffinMan10110** [comment]
@bhelx I reverted the change in question back to what it was.

**Comment by bhelx** [comment]
Thanks @MuffinMan10110 !

**Pull request merged**